src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6998,6 +6998,60 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 tmp = ASR::make_LogicalConstant_t(al, loc, is_lower,
                         ASRUtils::TYPE(ASR::make_Logical_t(al, loc, 4)));
                 return;
+            } else if (attr_name == "isalpha") {
+                /*
+                * Specification:
+                * Return True if all characters in the string are alphabetic (letters), and False otherwise.
+                */
+                bool is_alpha = true;
+                for (auto &i : s_var) {
+                    if (!std::isalpha(i)) {
+                        is_alpha = false;
+                        break;
+                    }
+                }
+                tmp = ASR::make_LogicalConstant_t(al, loc, is_alpha,
+                        ASRUtils::TYPE(ASR::make_Logical_t(al, loc, 4)));
+                return;
+            } else if (attr_name == "istitle") {
+                /*
+                * Specification: Return True if the string is in title case, where the first
+                * letter of each word is capitalized and the rest are lowercase. Return False
+                * if the string is empty or does not meet the title case criteria.
+                */
+                bool is_title = true;
+                int length = s_var.length();
+                if (length == 0) {
+                    is_title = false;
+                }
+
+                bool word_start = true;  
+                bool only_whitespace = true;
+
+                for (auto &ch : s_var) {
+                    if ((ch == ' ' || ch == '\t' || ch == '\n') && word_start) {
+                        continue; 
+                    } else if (std::isalpha(ch) && (std::isupper(ch))) {
+                        only_whitespace = false;
+                        if (word_start) {
+                            word_start = false;
+                        } else {
+                            is_title=false;  
+                        }
+                    } else if (std::isalpha(ch) && (std::islower(ch))) {
+                        only_whitespace = false;
+                        if (word_start) {
+                            is_title=false;  
+                        }
+                        word_start = false;
+                    } else {
+                        word_start = true;
+                    }
+                }
```

Comment:
I am slightly concerned about the logic we use here. I wonder if it could be simplified.

Reply:
Sure, it can be , I will try tomorrow morning.

Reply:
Hi @Shaikh-Ubaid, actually we handling 3 flags in here that's why we have complex branching 
if ((ch == ' ' || ch == '\t' || ch == '\n') && word_start) {
                        continue; 
 ```
 we need the only_whitespace flag to tack if all character inside the string is whitespace only so that we can mark the below example as false 
 ```
 (lp) ➜  lpython git:(redo_ismethods) python3                        
Python 3.10.2 | packaged by conda-forge | (main, Mar  8 2022, 15:52:47) [Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> "   ".istitle()
False
>>> "   H".istitle()
True
>>> "   \n".istitle()
False
>>> "   \t".istitle()
False
>>> "   \r".istitle()
False
```
we need word_start to mark it true if a non-alpha char is found so that we can check if the next word is the title case
```
(lp) ➜  lpython git:(redo_ismethods) ✗ python3
Python 3.10.2 | packaged by conda-forge | (main, Mar  8 2022, 15:52:47) [Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> "Hello\rRock".istitle()
True
>>> "Hello\nRock".istitle()
True
>>> "Hello'Rock".istitle()
True
```
and we need is_ttile to store the response 

I think we can somehow merge the 2 branches into one but for simplicity, we should keep these separate
I have added comments to the code